calyx-opt/src/passes/static_promotion.rs

```diff
@@ -190,6 +197,44 @@ impl Named for StaticPromotion {
 }
 
 impl StaticPromotion {
+    // Looks through ctx to get the given command line threshold.
+    // Default threshold = 1
+    fn get_threshold(ctx: &ir::Context) -> u64
```

Comment:
Does the `get_opts` method not work for this? It should be used for parsing simple options.